medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -112,8 +117,7 @@ pub async fn get_blob_name_from_content_id(
             return Ok(result.metadata_storage_name);
         }
     }
-    Err(anyhow!(format!(
-        "Cannot find document with content ID {}",
+    Err(anyhow!(errors::DocumentNotFoundInIndex::for_content_id(
```

Comment:
I think we can and should unit test this. It'll require an adapter for the `AzureSearchClient`, but it will be worth it to test that the right error is representing the _blob not found_ scenario.

Reply:
we can defer this too